Add config option for custom image tag format

detjensrobert opened the pull request:
Instead of hardcoding the expected challenge container image tag format, expose that to the user with a new config option `registry.tag_format`.

Some registries (AWS....) make it difficult to create separate image repositories for each different container. This lets that user change the tag format that beavercds is using, e.g. to something that uses many tags inside one repository.

detjensrobert committed:
Add new tag_format field for images to rcds.yaml config

Signed-off-by: Robert Detjens <github@detjens.dev>

detjensrobert committed:
Use tag_format field during build process

Signed-off-by: Robert Detjens <github@detjens.dev>

detjensrobert committed:
actually slugify challenge name as documented

Signed-off-by: Robert Detjens <github@detjens.dev>

detjensrobert committed:
Use image tag format template for access check image

Signed-off-by: Robert Detjens <github@detjens.dev>

detjensrobert committed:
Fix tag_format envvar override

Signed-off-by: Robert Detjens <github@detjens.dev>

KekoaM commented on src/configparser/config.rs:
```diff
@@ -60,10 +61,43 @@ struct RcdsConfig {
+    ///
+    /// Format: Jinja-style double-braces around field name (`{{ field_name }}`)
+    ///
+    /// Default: `"{{domain}}/{{challenge}}-{{container}}:{{profile}}"`
```
may want to indicate one or more common providers that the default works with (aws, gcp, etc.) and/or offer an alternative for other common providers
not a blocking additon

detjensrobert replied:
good idea, done

KekoaM reviewed (approved):
Overall implementation looks good, minor nit on documentation (add example for aws vs non-aws format)

detjensrobert committed:
Add tag_format example for AWS ECR

Signed-off-by: Robert Detjens <github@detjens.dev>

detjensrobert committed:
Gracefully report error in tag_format template

`render!()` panics when a strict-mode error occurs. Add a small helper fn
to render template strings without the macro and gracefully return errors.

Signed-off-by: Robert Detjens <github@detjens.dev>
